### DISCUSSION TITLE
Securing tasmota MQTT using TLS subject alternative name

### DISCUSSION THREAD

**Discussion by jsprenkle** [arendst/Tasmota#21907] [Q&A] [answered]
Can tasmota TLS be made to work with a static IP address (not an FQDN)?
My MQTT broker isn't in DNS. I use static IP's  (so DHCP doesn't shuffle my devices)

[This site](https://nodeployfriday.com/posts/self-signed-cert/) says it's possible, but I'm not sure Tasmota will honor the SAN field:
"To get the self signed cert to work with just an IP (not a domain name), we will specify a subject alternative name (SAN) for the IP."

Is there an alternative?

Thanks

**Comment by Jason2866** [comment]
Have you tried with a static IP?

**Reply by jsprenkle** [reply]
I'm building it one step at a time to make debugging easier.

**Tested**
- I configured the mosquitto server
- Used MQTT explorer to connect to it securely.
- Tried the command line utilities to confirm it works.
**Using mosquitto_pub with a static IP gives a generic "TLS Error".
Changing the IP address to the name of the server and leaving the rest the same does work, and is confirmed in explorer.**
All my research says it will not work without my server being in the DNS.

**Not tested yet:**
- recompile Tasmota with TLS enabled and paste in my self signed Certificate Authority Cert.
- Unknown if a work around is needed for the FQDN issue.

**Comment by s-hadinger** [comment, marked as answer]
I have already answered on Discord, not sure it was you asking. 

As per TLS specification, ip addresses are not accepted. Your certificate must match the domain name

**Reply by jsprenkle** [reply]
Thank you. I'd rather not stand up a DNS server just for this. I'll see if mDNS is working in the newer versions of Tasmota. I read some time ago it wasn't.

**Reply by s-hadinger** [reply]
I doubt that mDNS domain names are valid for TLS either.

**Reply by jsprenkle** [reply]
Thanks for chiming in. Saved me a lot of work and frustration trying to do something that will not work
